service/src/main/resources/HTMLTemplate.html

```diff
@@ -1066,44 +1074,52 @@ <h3 class="heading heading--two">Respondent
     <div class="data-list__row">
       <dt class="data-list__title">Current address</dt>
       <dd class="data-list__definition">
-        {% if case_data.respondents is not empty and
-        case_data.respondents.firstRespondent is not empty and
-        case_data.respondents.firstRespondent.address is not empty %}
-          {% if case_data.respondents.firstRespondent.address.AddressLine1 is not empty %}
-            <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.AddressLine1 }}</span>
-          {% endif %}
-          {% if case_data.respondents.firstRespondent.address.AddressLine2 is not empty %}
-            <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.AddressLine2 }}</span>
-          {% endif %}
-          {% if case_data.respondents.firstRespondent.address.AddressLine3 is not empty %}
-            <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.AddressLine3 }}</span>
-          {% endif %}
-          {% if case_data.respondents.firstRespondent.address.PostTown is not empty %}
-            <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.PostTown }}</span>
-          {% endif %}
-          {% if case_data.respondents.firstRespondent.address.PostCode is not empty %}
-            <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.PostCode }}</span>
-          {% endif %}
-          {% if case_data.respondents.firstRespondent.address.County is not empty %}
-            <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.County }}</span>
-          {% endif %}
-          {% if case_data.respondents.firstRespondent.address.Country is not empty %}
-            <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.Country }}</span>
+        {% if case_data.respondents is not empty and case_data.respondents.firstRespondent is not empty %}
```

Comment:
The way it is implemented now hides the fact that address was not provided when details were marked confidential. Shouldn't we check whether address exists first and then if exists check whether it is confidential / should be hidden?

AC says

```
Given the address of a respondent is added in the application
And the respondents address field is marked as confidential by saying yes to the question Do you need contact details hidden from other parties? (Optional)
When the application document is generated
Then the address of the respondent will read 'Confidential' on the generated pdf document
And the telephone number of the respondent will read 'Confidential' on the generated pdf document
```

which to me sounds like an action should be taken only when address exists. If not regardless whether confidential or not we should display `-`

Reply:
good spot, that's a better way of implementing